external-crates/move/crates/move-compiler/src/naming/translate.rs

```diff
@@ -1228,17 +1423,13 @@ fn call_args(context: &mut Context, sp!(loc, es): Spanned<Vec<E::Exp>>) -> Spann
 }
 
 fn exps(context: &mut Context, es: Vec<E::Exp>) -> Vec<N::Exp> {
-    es.into_iter().map(|e| exp_(context, e)).collect()
-}
-
-fn exp(context: &mut Context, e: E::Exp) -> Box<N::Exp> {
-    Box::new(exp_(context, e))
+    es.into_iter().map(|e| *exp(context, Box::new(e))).collect()
 }
 
-fn exp_(context: &mut Context, e: E::Exp) -> N::Exp {
+fn exp(context: &mut Context, e: Box<E::Exp>) -> Box<N::Exp> {
```

Comment:
Same comment here about changing this to take box exps. Maybe it was an attempt to refactor stuff? It's a lot of churn for no apparent gain, especially with the stacker stuff.

Reply:
I guess I just view it as more "correct", especially in debug builds. Even with a larger stack size, its more to keep on the stack with each recursive call.
I can revert if you want still